src/Server/Environment.php

```diff
@@ -88,13 +88,20 @@ public static function setDefault($name, $value)
     public function parse($value)
     {
         if (is_string($value)) {
-            if (preg_match_all('/\{(.+?)\}/', $value, $matches)) {
-                foreach ($matches[1] as $name) {
-                    $value = str_replace('{' . $name . '}', $this->get($name), $value);
-                }
-            }
+            $value = preg_replace_callback('/\{\{\s*(\w+)\s*\}\}/', [$this, '_parse'], $value);
         }
 
         return $value;
     }
+    
+    /**
+     * Replace env values callback for parse
+     *
+     * @param array $matchs
+     * @return mixed
+     */
+    private function _parse($matchs)
```

Comment:
Please do not user under scope `_` for private methods. Rename it to `parseCallback`.